### DISCUSSION TITLE
Can a parent container spring match the duration of a child multi-stage spring?

### DISCUSSION THREAD

**Discussion by samselikoff** [pmndrs/react-spring#1328] [Support]
Hi there! Done some searching but couldn't find anything.

Is it possible to "back out" or derive the duration from a given spring config?

The reason I'm asking, is that I'm making a "Container" spring that wraps an inner transition spring. The inner spring is a multi-stage transition, and I want the Container's spring config to match the total duration of the complete multi-stage inner animation.

I know there's a `duration` prop but that seems to change the nature of the animations.

I'm new to the lib so any pointers would be appreciated! Thanks for working on this awesome lib 👍

**Comment by harshagarwalnyu** [comment]
You can "back out" spring duration using react-spring's `easings` or by switching the parent to `duration`-based config. Here's the pragmatic approach:

**Option 1 — Use `duration` config on both (simplest)**
```js
const STAGE_1_MS = 400;
const STAGE_2_MS = 300;
const TOTAL_MS = STAGE_1_MS + STAGE_2_MS;

// Child multi-stage spring
const childSpring = useSpring({
  to: async (next) => {
    await next({ x: 100, config: { duration: STAGE_1_MS } });
    await next({ y: 50, config: { duration: STAGE_2_MS } });
  },
});

// Parent matches total duration
const parentSpring = useSpring({
  height: isOpen ? 200 : 0,
  config: { duration: TOTAL_MS, easing: easings.easeInOutQuad },
});
```

**Option 2 — `onRest` chain (react to completion, not predict duration)**
```js
const [parentReady, setParentReady] = useState(false);

const childSpring = useSpring({
  to: async (next) => {
    setParentReady(false);
    await next({ x: 100, config: { duration: 400 } });
    await next({ y: 50, config: { duration: 300 } });
    setParentReady(true);  // signal parent that child finished
  },
});

const parentSpring = useSpring({
  height: parentReady ? 200 : 100,
  config: { tension: 200, friction: 25 },
});
```

**Option 3 — Compute spring duration with `react-spring`'s internal math**

For physics-based springs, duration ≈ `(-Math.log(0.001) / damping) * 1000`. But springs are asymptotic — they never truly "end". Use `precision` config to control when `onRest` fires:
```js
config: { tension: 200, friction: 25, precision: 0.01 }
```

**Recommendation:** use `duration`-based config for both if you need exact synchronization. Physics springs are inherently imprecise about end-time. The `duration` approach gives you deterministic timing at the cost of the "natural" spring feel — but for matched animations, determinism wins.
